examples/Atlas/+atlasControllers/@InstantaneousQPController/setupAndSolveQP.m

```diff
@@ -0,0 +1,355 @@
+function [y, qdd, info_fqp, supp,...
+          alpha, Hqp, fqp, Aeq, beq, Ain, bin,lb,ub] = setupAndSolveQP(obj, t, q, qd, qp_input, contact_sensor)
+% Construct the quadratic control problem for Atlas and solve it, returning
+% torques and accelerations. This file currently represents most of the inner
+% loop of atlasControllers.AtlasQPController.output(), but extracted out into
+% its own function. It also mirrors the functionality of instantaneousQPControllermex.cpp
+% @param t current time (s)
+% @param q, qd current robot pose and velocity
+% @param qp_input QPInput2D describing the current plan state
+% @param contact_sensor num_bodies vector indicating whether the given body has detected contact force
+
+params = obj.param_sets.(qp_input.param_set_name);
+r = obj.robot;
+ctrl_data = obj.controller_data;
+
+% Compute desired body accelerations and costs
+all_bodies_vdot = struct('body_id', cell(1,length(qp_input.body_motion_data)),...
+                         'body_vdot', cell(1,length(qp_input.body_motion_data)),...
+                         'params', cell(1,length(qp_input.body_motion_data)));
+fun = fcompare(@atlasControllers.bodyMotionControl,@instantaneousBodyMotionControlmex);
+for j = 1:length(qp_input.body_motion_data)
+  body_id = qp_input.body_motion_data(j).body_id;
+  [body_des, body_v_des, body_vdot_des] = evalCubicSplineSegment(t - qp_input.body_motion_data(j).ts(1), qp_input.body_motion_data(j).coefs);
+  body_vdot = fun(r, [q; qd], body_id,...
+                                body_des, body_v_des, body_vdot_des, params.body_motion(body_id));
+  all_bodies_vdot(j).body_id = body_id; 
+  all_bodies_vdot(j).body_vdot = body_vdot;
+  all_bodies_vdot(j).params = params.body_motion(body_id);
+end
+
+% Find the active set of supports
+mask = getActiveSupportsmex(obj.mex_ptr, [q; qd], qp_input.support_data, contact_sensor, params.contact_threshold, obj.default_terrain_height);
+supp = qp_input.support_data(logical(mask));
+
+% Run PD on the desired configuration
+qddot_des = obj.wholeBodyPID(t, q, qd, qp_input.whole_body_data.q_des, params.whole_body);
```

Comment:
nevermind, i'm ok with how it's implemented below (didn't realize the method was also owned by this class)